.github/workflows/build-and-publish.yml

```diff
@@ -29,6 +29,49 @@ jobs:
       - name: Set package version
         run: |
           mvn versions:set -DnewVersion="1.0.$(date +"%Y%m%d%H%M%S")"
+
+      - name: Check Signing Key Validity
+        env:
+          GPG_PRIVATE_KEY: ${{ secrets.GPG_PRIVATE_KEY }}
+        run: |
```

Comment:
decent amount of bash going on in this step, perhaps we could move this to `pay-ci` as a custom action to reduce cognitive complexity of the workflow?

Reply:
It's in it's own workflow now, I added it into the unit tests to check it again and this is the failed check seeing the expired key https://github.com/alphagov/pay-java-commons/actions/runs/4786491115/jobs/8510532253?pr=647

Reply:
okay, can live with that. LG